stactools_landsat/stactools/landsat/utils.py

```diff
@@ -0,0 +1,141 @@
+import datetime
+
+import dateutil
+import rasterio
+from pystac import Item, Link
+from rasterio import RasterioIOError
+from shapely.geometry import box, mapping, shape
+
+
+def _parse_date(in_date: str) -> datetime.datetime:
+    """
+    Try to parse a date and return it as a datetime object with no timezone
+    """
+    dt = dateutil.parser.parse(in_date)
+    return dt.replace(tzinfo=datetime.timezone.utc)
+
+
+def transform_mtl_to_stac(metadata: dict) -> Item:
+    """
+    Handle USGS MTL as a dict and return a STAC item.
+
+    NOT IMPLEMENTED
+
+    Issues include:
+        - There's no reference to UTM Zone or any other CRS info in the MTL
+        - There's no absolute file path or reference to a URI to find data.
+    """
+    LANDSAT_METADATA = metadata["LANDSAT_METADATA_FILE"]
+    product = LANDSAT_METADATA["PRODUCT_CONTENTS"]
+    projection = LANDSAT_METADATA["PROJECTION_ATTRIBUTES"]
+    image = LANDSAT_METADATA["IMAGE_ATTRIBUTES"]
+    proessing_record = LANDSAT_METADATA["LEVEL2_PROCESSING_RECORD"]
+
+    scene_id = product["LANDSAT_PRODUCT_ID"]
+
+    xmin, xmax = float(projection["CORNER_LL_LON_PRODUCT"]), float(
+        projection["CORNER_UR_LON_PRODUCT"])
+    ymin, ymax = float(projection["CORNER_LL_LAT_PRODUCT"]), float(
+        projection["CORNER_UR_LAT_PRODUCT"])
```

Comment:
I don't think this is correct for lon/lat coordinates. Because the image is tilted, isn't `ymin` in the lower right corner?

Reply:
These coordinates actually are the bounding box coordinates, not the data footprint so the rotation doesn't factor in.

Reply:
Yeah I clearly missed that the first time 😄